### ISSUE TITLE
Adding Xamarin-Sidebar to an existing project

### ISSUE THREAD

**Issue by mguillen** [jdehlin/Xamarin-Sidebar#11]
Could you list down the steps on how I can add this library to a project? Thank you very much!

**Comment by jdehlin** [comment]
If you are using Xamarin you can install the component from http://components.xamarin.com/view/SidebarNavigation.

If you don't want to use the component you have two options. First is checkout this repository then run and build it. This will generate a dll in the bin folder which you can copy to your project and reference. The other option is to just create a cs file in your project and copy the contents of https://github.com/jdehlin/Xamarin-Sidebar/blob/master/Xamarin-Sidebar/SidebarController.cs into your file.

**Comment by mguillen** [comment]
I got it Sir @jdehlin, my next problem is how do I use this class when I'm using Storyboard. I can't make it work, I need your help.

**Comment by pilucdes** [comment]
How it work with storyboard : 
1. All view can be done in storyboard but you can't add segue from storyboard since the navigation controller is not in storyboard but in code behind 
1. If you want to push you will have to instance your view from the code "Storyboard.InstantiateViewController" and push manually from the navigation controller that you have make an instance in the root view and hold reference in appdelegate(well this is how i do it)

This is where all start create a first view that will be your root in the storybard. Set it as rootview (starting point) in the storyboard. This view is empty since it's the container

Here is my rootview  ViewDidLoad function : 

```
        //Set Navigation controller reference so we can use it everywhere
        AppDelegate appD = UIApplication.SharedApplication.Delegate as AppDelegate;
        appD.GETSET_NavController = new UINavigationController();


        VCConnection vConnect = (VCConnection)Storyboard.InstantiateViewController("VCConnection");
        VCLeftMenu vLeftMenu = (VCLeftMenu)Storyboard.InstantiateViewController("VCLeftMenu");
        appD.GETSET_NavController.PushViewController(vConnect,false);
        appD.GETSET_SidebarController = new SidebarController(this, appD.GETSET_NavController, vLeftMenu);
        appD.GETSET_SidebarController.MenuWidth = 250;
        appD.GETSET_SidebarController.ReopenOnRotate = false;
        appD.GETSET_SidebarController.IsOpening += appD.GETSET_Menu.UpdatePicturesToSynch;
```

VConnection is my default view to be shown in the content.
VCLeftMenu is the view that will be shown in the  sidebarMenuNavController
 appD.GETSET_NavController is the navigationController instanced

I also hold a reference to the sidebarController in case i need it some where...

Basically, in this case, the storyboard use is to create view using a UI and apply constraint to it that's pretty much it. 

When i want to change the content view i simply do this : 

```
      UIViewController vcSupport = (UIViewController)Storyboard.InstantiateViewController("VCSupport");
        GET_APPD.GETSET_NavController.PushViewController(vcSupport,true);
```

I instance from storyboard then i push with the reference navigation controller in the appdelegate

**Comment by pilucdes** [comment]
Look my answer above !

**Comment by jdehlin** [comment]
Thanks for answering how to use it with Storyboard, @pilouk !

**Comment by mguillen** [comment]
I did the same way and made it work while waiting for your reply. But it's really good to know that we could have a reference to the Navigation and Sidebar Controllers. Thank you very much @pilouk and @jdehlin
